openedx/core/djangoapps/schedules/config.py

```diff
@@ -19,3 +24,49 @@
     'course_update_show_unsubscribe',
     __name__
 )
+
+# This experiment waffle is supporting an A/B test we are running on sending course updates from an external service,
+# rather than through platform and ACE. See ticket AA-661 for more information.
+# Don't use this flag directly, instead use the `set_up_external_course_updates` and `query_external_updates` methods
+# below. We save this flag decision at enrollment time and don't change it even if the flag changes. So you can't just
+# directly look at flag result.
+_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID = 18
+_EXTERNAL_COURSE_UPDATES_FLAG = ExperimentWaffleFlag(WAFFLE_FLAG_NAMESPACE, 'external_updates', __name__,
+                                                     experiment_id=_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID,
+                                                     use_course_aware_bucketing=False)
+
+
+def set_up_external_course_updates(user, course_key):
+    """
+    Returns and stores whether a user should be getting the "external course updates" experience.
+
+    Basically just a wrapper around our experiment waffle flag, but only buckets users that directly enrolled (rather
+    than users enrolled by staff), for technical "waffle-flags-can-only-get-the-user-from-the-request" reasons
+
+    This saves the decision in experiment data tables. It is also idempotent and will not change after the first
+    call for a given user/course, regardless of how the waffle answer changes.
+    """
+    request = get_current_request()
+    user_is_valid = request and hasattr(request, 'user') and request.user.id and request.user.id == user.id
+    experiment_on = _EXTERNAL_COURSE_UPDATES_FLAG.is_experiment_on(course_key)
+    if user_is_valid and experiment_on:
+        # Don't send tracking info as it might differ from our saved value, and we already send the bucket in
+        # enrollment segment events.
+        bucket = _EXTERNAL_COURSE_UPDATES_FLAG.get_bucket(course_key, track=False)
+    else:
+        bucket = -1  # a special value meaning to ignore this enrollment for analytics purposes
+
+    data, _created = ExperimentData.objects.get_or_create(experiment_id=_EXTERNAL_COURSE_UPDATES_EXPERIMENT_ID,
+                                                          user_id=user.id, key=str(course_key),
+                                                          defaults={'value': str(bucket)})
+    return int(data.value)
+
+
+def query_external_updates(user_id, course_id):
+    """
+    Returns a queryset filtering only the "external course updates" experience for the given user/course.
```

Comment:
could you clarify this comment?

Reply:
I'm not sure how honestly. I must be too in the weeds to see what is likely to be confusing about it.

Ah... would it help to add "this is designed for use as a subquery in a larger queryset, but could also be used to spot-check whether a user is in the external experience for a given course"?

Reply:
Maybe something like
Returns a queryset that indicates whether the user should see external course updates for a given course

Reply:
I can wordsmith it up a bit and add some context, sure.

Though the plan in general is that this code is not long lived, so hopefully no one else ever actually reads this docstring.

Reply:
good point, i guess if this code is an experiment and will be deleted soon feel free to ignore all my comments about your wording, sorry about that

Reply:
Naw, I expanded the docstring a bit - let me know if it's clearer now?